gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspAATask.kt

```diff
@@ -307,17 +321,32 @@ abstract class KspAAWorkerAction : WorkAction<KspAAWorkParameter> {
             languageVersion = gradleCfg.languageVersion.get()
             apiVersion = gradleCfg.apiVersion.get()
 
-            logger = kspGradleLogger
-
             processorOptions = gradleCfg.processorOptions.get()
             allWarningsAsErrors = gradleCfg.allWarningsAsErrors.get()
 
             jvmTarget = gradleCfg.jvmTarget.get()
             jvmDefaultMode = gradleCfg.jvmDefaultMode.get()
         }.build()
+        val byteArrayOutputStream = ByteArrayOutputStream()
+        val objectOutputStream = ObjectOutputStream(byteArrayOutputStream)
+        objectOutputStream.writeObject(kspConfig)
 
         val exitCode = try {
-            KotlinSymbolProcessing(kspConfig).execute()
+            val kspLoaderClass = isolatedClassLoader.loadClass(KSPLoader::class.java.canonicalName)
+            val kspLoader = kspLoaderClass.newInstance()
```

Comment:
nit: Let's make `loadAndRunKSP` a static function and use `@JvmStatic` to force its name. Then we can avoid instantiating `KspLoader`.